src/braintree/initBraintreePayPalButtons.ts

```diff
@@ -0,0 +1,310 @@
+import { getTotals, loadJS } from 'src/utils';
+import { showPaymentMethodTypes } from 'src/variables';
+import { braintreeOnLoadClient } from './braintreeOnLoadClient';
+import { getBraintreeJsUrls } from './getBraintreeJsUrls';
+import { getBraintreeClient } from './manageBraintreeState';
+
+// TODO move this interface to frontend library
+export interface IExpressPayBraintreePaypal {
+    is_paylater_enabled: boolean;
+    is_test: 0 | 1;
+    properties: {
+        public_id: string;
+        commit: boolean;
+    };
+    tokenization_key: string;
+    type: 'braintree-paypal';
+    validation_schema: unknown[];
+}
+
+interface IEstimateShippingLinesResponse {
+    // Yes this is the correct key
+    'shipping options': IShippingLine[];
+    application_state: IApplicationState;
+}
+
+const handleCreateOrder = (paypalCheckout: IBraintreePaypalCheckoutInstance) => async (): Promise<string> => {
+    const totals = getTotals();
+
+    return paypalCheckout.createPayment({
+        currency: getCurrency().iso_code,
+        flow: 'checkout',
+        amount: totals.totalAmountDue / 100,
+        intent: 'authorize',
+        requestBillingAgreement: true,
+        enableShippingAddress: getLineItems().some(li => li.product_data.requires_shipping),
+        shippingOptions: [],
+    });
+};
+
+const handleApprove = (
+    paypalCheckout: IBraintreePaypalCheckoutInstance,
+    payment: IExpressPayBraintreePaypal,
+) => async (
+    data: OnApproveData,
+    actions: OnApproveActions,
+): Promise<void> => {
+    /* istanbul ignore next */
+    if (!actions.order) {
+        displayError('There was an unknown error while processing your payment.', 'payment_gateway', 'unknown_error');
+        return;
+    }
+
+    const [result, order] = await Promise.all([
+        paypalCheckout.tokenizePayment({
+            payerId: data.payerID ?? '',
+            billingToken: data.billingToken ?? undefined,
+            paymentId: data.paymentID ?? undefined,
+            vault: false
+        }),
+        actions.order.get(),
+    ]);
+
+    const countries = getOrderInitialData().country_info;
+    const customer = getCustomer();
+    const shippingAddress = result.details.shippingAddress;
+    const billingAddress = result.details.billingAddress;
+    const shippingCountry = shippingAddress ? countries.find(c => c.iso_code === shippingAddress.countryCode) : null;
+    const billingCountry = countries.find(c => c.iso_code === billingAddress.countryCode);
+    /* istanbul ignore next */
+    const shippingProvince = shippingCountry?.provinces.find(p => p.iso_code === shippingAddress?.state);
+    /* istanbul ignore next */
+    const billingProvince = billingCountry?.provinces.find(p => p.iso_code === billingAddress.state);
+    const shippingOption = order.purchase_units[0].shipping?.options?.find(o => o.selected);
+    
+    /* istanbul ignore next */
+    if (!billingProvince || !billingCountry) {
```

Comment:
what if `shippingProvince` or `shippingCountry` are undefined? Do we need to throw an error, or is it okay to continue with empty values?